azurekvp/src/tracing.rs

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/// This module provides core functionality for handling telemetry tracing.
+/// related to azure-init's interaction with Hyper-V KVP (Key-Value Pair) storage.
+///
+/// # Constants
+/// - `HV_KVP_EXCHANGE_MAX_KEY_SIZE`: Defines the maximum key size for KVP exchange.
+/// - `HV_KVP_EXCHANGE_MAX_VALUE_SIZE`: Defines the maximum value size for KVP exchange.
+/// - `HV_KVP_AZURE_MAX_VALUE_SIZE`: Maximum value size before splitting into multiple slices.
+///
+use chrono::{DateTime, Utc};
+use nix::fcntl::{flock, FlockArg};
+use opentelemetry::{global, sdk::trace as sdktrace, trace::TracerProvider};
+use std::fs::OpenOptions;
+use std::io::ErrorKind;
+use std::io::{Error, Write};
+use std::os::fd::AsRawFd;
+use std::os::unix::fs::MetadataExt;
+use std::path::Path;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+use uuid::Uuid;
+
+const HV_KVP_EXCHANGE_MAX_KEY_SIZE: usize = 512;
+const HV_KVP_EXCHANGE_MAX_VALUE_SIZE: usize = 2048;
+const HV_KVP_AZURE_MAX_VALUE_SIZE: usize = 1024;
+const EVENT_PREFIX: &str = "AZURE_INIT";
+
+pub fn initialize_tracing() -> sdktrace::Tracer {
+    let provider = sdktrace::TracerProvider::builder().build();
+    global::set_tracer_provider(provider.clone());
+    provider.tracer("azure-kvp")
+}
+
+pub fn handle_span<S>(
```

Comment:
Since function is used exclusively for the `on_close` handler, I would recommend just moving the contents into that function and then attaching your helper functions as methods on your `EmitKVPLayer` struct